Avoid infinite loops in WDFreehandTool (2 issue screenshots).

rhcad opened the pull request:
[Issue screenshot 1](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad-err1.png)
[Issue screenshot 2](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad-err2.png)

rhcad committed:
Avoid infinite loops in WDFreehandTool.
[Issue screen shot 1](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad-err1.png)
[Issue screen shot 2](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad-err2.png)

sprang commented:
Is the problem easy to reproduce? It would be be great to figure out where the NaNs get introduced. 

In +[WDBezierNode bezierNodeWithInPoint:anchorPoint:outPoint:] you're not testing the anchor point.

rhcad committed:
Avoid infinite loops in WDFreehandTool.

NaN point may be added in FitCurve, FitCubic, so return nil in pathFromSegments.
Cancel the changes yesterday in WDBezierNode.m to avoid unknown issue about null pointer.

rhcad commented:
I have added a new submit just now to perfect the possible issus:
1. NaN point may be added in FitCurve and FitCubic, so return nil in pathFromSegments.
2. Cancel the changes yesterday in WDBezierNode.m to avoid unknown issue about null pointer.

It's easy to reproduce: draw some unclosed curve which like arc ( [see the screenshot](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad_issue/howtofire.png) ), then the infinite loops problem appeared.

The call stack about pathFromSegments:
![nan_outside](https://f.cloud.github.com/assets/558922/1855932/26fb1cfe-773e-11e3-9bb1-ea418776b4b7.png)
I have fixed the function via verifying the first segment or the problem will appear.

The call stack about FitCubic called by smoothPathForPoints which will call pathFromSegments:
![nan_inside](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad_issue/NaN_inside.png)

In addition, there may be a problem in GenerateBezier: SegLength and epsilon are zero when the above problem appeared. 
![GenerateBezierMayIssue](https://raw.github.com/rhcad/touchvg-doc/gh-pages/images/Inkpad_issue/maybe_issue.png)

Thanks!
Zhang Yungui

sprang commented:
I believe commit bf6a8d52053a367963cb31388f1aae08437544eb resolved the root cause of this issue, so this patch should no longer be necessary.
